apps/sms/views/shared/test/unit/activity_handler_test.js

```diff
@@ -45,71 +26,41 @@ require('/views/shared/test/unit/mock_utils.js');
 require('/views/shared/js/activity_handler.js');
 
 var mocksHelperForActivityHandler = new MocksHelper([
+  'ActivityClient',
+  'ActivityShim',
   'Attachment',
-  'Contacts',
   'Draft',
   'Drafts',
   'MessageManager',
-  'NotificationHelper',
-  'Notify',
-  'Settings',
-  'SettingsURL',
-  'SilentSms',
-  'SMIL',
-  'Threads',
-  'Utils',
   'Navigation',
-  'ActivityShim',
-  'ActivityClient'
+  'Settings',
+  'Utils'
 ]).init();
 
 suite('ActivityHandler', function() {
   mocksHelperForActivityHandler.attachTestHelpers();
 
-  var realSetMessageHandler;
-  var realWakeLock;
-  var realMozApps;
-  var realMozL10n;
-
-  var isDocumentHidden;
+  var navigatorMocks = new Map([
```

Comment:
I'm fine with the changes since the original navigator mocks did a lot of similar action repeatedly and we can rewire it in simpler way apparently. Just a question that you didn't preserve the original navigator API(and no way to resume obviously). I think for the most of the case it should work since almost all the navigatior API will not be accessible in the test, and it seems meaningless to resume while suiteTeardown. So just want to confirm that you did this for specific reason.